### DISCUSSION TITLE
zos and automatic bandwidth limiting/metering/billing

### DISCUSSION THREAD

**Discussion by delandtj** [threefoldtech/zos#1229] [Ideas]
One of the hot potatoes that don't get handled in zos because of:
  - hard
  - no clue
  - complex

is:

How do we make sure that network and ip transit abusers get 
  - squeezed in as well bandwidth as in pps
  - billed accordingly
  - get a proper view on how much data is effectively transmitted

over:
  - his/her private network (how do we bill that?)
  - towards/from the Internet itself

and:
  - how to alert the farmer
  - how to allow the farmer to override default transfer policies for a reservation
  - how to set default policies.
 
Thing is:
When we'll get a little exposure, we __will__ have more abusers than regular users. (that is not an assumption, but a certitude)

Basically, for the grid itself with the webgateways, we'll have to start thinking about that too.

**Comment by muhamadazmy** [comment]
This one probably will be impossible for grid 2.6 since billing is done completely on the explorer side. It's possible for grid 3 though since the node is the one that collects metrics about the user consumption and bill accordingly.
